File: test/extensions/language/choice_types/cpp/ExpressionSelectorChoiceTest.cpp
#include "choice_types/expression_selector_choice/ExpressionSelectorChoice.h"
#include "gtest/gtest.h"

namespace choice_types
{
namespace expression_selector_choice
{

TEST(ExpressionSelectorChoiceTest, field8)
{
    const uint16_t selector = 0;
    ExpressionSelectorChoice expressionSelectorChoice;
    expressionSelectorChoice.initialize(selector);
    const uint8_t value8 = 0x7F;
    expressionSelectorChoice.setField8(value8);
    ASSERT_EQ(value8, expressionSelectorChoice.getField8());
    ASSERT_EQ(8, expressionSelectorChoice.bitSizeOf());
}

TEST(ExpressionSelectorChoiceTest, field16)
{
    const uint16_t selector = 1;
    ExpressionSelectorChoice expressionSelectorChoice;
    expressionSelectorChoice.initialize(selector);
    const uint16_t value16 = 0x7F7F;
    expressionSelectorChoice.setField16(value16);
    ASSERT_EQ(value16, expressionSelectorChoice.getField16());
    ASSERT_EQ(16, expressionSelectorChoice.bitSizeOf());
}

TEST(ExpressionSelectorChoiceTest, field32)
{
    const uint16_t selector = 2;
    ExpressionSelectorChoice expressionSelectorChoice;
    expressionSelectorChoice.initialize(selector);
    const uint32_t value32 = 0x7F7F7F7F;
    expressionSelectorChoice.setField32(value32);
    ASSERT_EQ(value32, expressionSelectorChoice.getField32());
    ASSERT_EQ(32, expressionSelectorChoice.bitSizeOf());
}

} // namespace expression_selector_choice
} // namespace choice_types

File: test/extensions/language/choice_types/python/ExpressionSelectorChoiceTest.py
import ChoiceTypes


class ExpressionSelectorChoiceTest(ChoiceTypes.TestCase):
    def testField8(self):
        expressionSelectorChoice = self.api.ExpressionSelectorChoice(0)
        value8 = 0x7F
        expressionSelectorChoice.field8 = value8
        self.assertEqual(value8, expressionSelectorChoice.field8)
        self.assertEqual(8, expressionSelectorChoice.bitsizeof())

    def testField16(self):
        value16 = 0x7F7F
        expressionSelectorChoice = self.api.ExpressionSelectorChoice(1, field16_=value16)
        self.assertEqual(value16, expressionSelectorChoice.field16)
        self.assertEqual(16, expressionSelectorChoice.bitsizeof())

    def testField32(self):
        expressionSelectorChoice = self.api.ExpressionSelectorChoice(2)
        value32 = 0x7F7F7F7F
        expressionSelectorChoice.field32 = value32
        self.assertEqual(value32, expressionSelectorChoice.field32)
        self.assertEqual(32, expressionSelectorChoice.bitsizeof())
